great_expectations/expectations/metrics/util.py

```diff
@@ -1090,7 +1100,7 @@ def get_unexpected_indices_for_multiple_pandas_named_indices(
         )
 
     domain_records_df_index_names: List[str] = domain_records_df.index.names
-    unexpected_indices: List[Union[int, str]] = list(domain_records_df.index)
+    unexpected_indices: List[tuple[int | str, ...]] = list(domain_records_df.index)
```

Comment:
@Kilo59 this is the method that only handles multiple named indices, so it should be a List of tuples, not a list of ints and strings :) Thanks for the discussion earlier.